8.x-2.x > devportal/8.x-2.x

lekob opened the pull request.

Commit:
Delete the old devportal_repo_sync module.

Commit:
Create module

Create new devportal_repo_sync module.

Commit:
Add config form and config schema.

Commit:
Create route and menu link for config form.

Commit:
Merge remote-tracking branch 'pronovix-devportal/8.x-2.x' into 8.x-2.x

Commit:
Merge branch '8.x-2.x' into 8.x-2.x_repo-sync

Commit:
Added create form, controller, links and routes.

Added a form through which an import can be created. Also added a
controller to show the imports, and routes and links for the form and
controller.

Commit:
Merge remote-tracking branch 'pronovix-devportal/8.x-2.x' into 8.x-2.x_repo-sync

Commit:
Added local action link to the create form.

Commit:
Created confirm delete form with route..

Commit:
Changes to controller and create form

Removed redundant dependency injection of the messenger service from the
controller and the create form. Changed how the rows are built in the
controller. Removed repository type select option from the create form.

Commit:
Minor changes to Client.php docblock, and info.yml.

tamasd commented on modules/repo_sync/config/install/devportal_repo_sync.config.yml:
```diff
@@ -0,0 +1 @@
+devportal_repo_sync:
```
This should have empty values for all possible keys.

tamasd commented on modules/repo_sync/config/schema/devportal_repo_sync.schema.yml:
```diff
@@ -0,0 +1,10 @@
+devportal_repo_sync.config:
+  type: config_object
+  label: 'Repo Sync'
+  mapping:
```
I think being able to configure the service URL would make sense.

tamasd commented on modules/repo_sync/devportal_repo_sync.routing.yml:
```diff
@@ -1,109 +1,42 @@
-entity.repo_account.collection:
-  path: '/admin/config/devportal/config/repo-sync/account'
+devportal_repo_sync.devportal_repo_sync_config_form:
+  path: '/admin/devportal/repo_sync/config'
```
`s/_/-/g`

tamasd commented on modules/repo_sync/devportal_repo_sync.routing.yml:
```diff
@@ -1,109 +1,42 @@
-entity.repo_account.canonical:
-  path: '/admin/config/devportal/config/repo-sync/account/{repo_account}'
+devportal_repo_sync.repo_sync_controller_content:
+  path: '/admin/devportal/repo_sync/content'
```
`s/_/-/g`

tamasd replied:
Also, shouldn't this be `/admin/devportal/repo-sync`?

tamasd commented on modules/repo_sync/devportal_repo_sync.routing.yml:
```diff
@@ -1,109 +1,42 @@
-entity.repo_account.add_form:
-  path: '/admin/config/devportal/config/repo-sync/account/add'
+devportal_repo_sync.devportal_repo_sync_create_form:
+  path: '/admin/devportal/repo_sync/create'
```
`s/_/-/g`

tamasd commented on modules/repo_sync/devportal_repo_sync.routing.yml:
```diff
@@ -1,109 +1,42 @@
-entity.repo_account.edit_form:
-  path: '/admin/config/devportal/config/repo-sync/account/{repo_account}/edit'
+devportal_repo_sync.devportal_repo_sync_delete_form:
+  path: '/admin/devportal/repo_sync/delete/{uuid}'
```
`s/_/-/g`

tamasd replied:
I think this should be `/admin/devportal/repo-sync/{uuid}/delete`

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+   */
+  public function content() {
+    $config = $this->config('devportal_repo_sync.config');
+    $client = new Client($config->get('uuid'), hex2bin($config->get('secret')), "http://service:8000");
```
The endpoint should be configurable, see my comment above.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+              '#links' => [
+                'view' => [
+                  'title' => $this->t('View'),
+                  'url' => Url::fromRoute('devportal_repo_sync.devportal_repo_sync_config_form'),
```
This link points to the wrong place.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+                ],
+                'edit' => [
+                  'title' => $this->t('Edit'),
+                  'url' => Url::fromRoute('devportal_repo_sync.devportal_repo_sync_config_form'),
```
This link points to the wrong place.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+      $this->messenger->addError($e->getMessage());
+    }
+
+    if (!empty($rows)) {
```
Unnecessary if.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+        '#description' => $this->t('Repository Synchronization settings overview.'),
+      ];
+    }
+    return $build ?? NULL;
```
I don't think that it is a good idea to return NULL here.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncController.php:
```diff
@@ -0,0 +1,71 @@
+      $result = json_decode(array_pop($result));
+
+      foreach ($result->items as $item) {
+        $rows[] = [
```
The `$rows` variable should be defined before the foreach, like this: `$rows = [];`

tamasd commented on modules/repo_sync/src/Form/RepoSyncCreateForm.php:
```diff
@@ -0,0 +1,113 @@
+      '#default_value' => '',
+      '#weight' => 5,
+    ];
+    $form['callback'] = [
```
This should be generated automatically by Drupal.

tamasd commented on modules/repo_sync/src/Form/RepoSyncCreateForm.php:
```diff
@@ -0,0 +1,113 @@
+        "Pattern" => $values["pattern"],
+        "Reference" => $values["reference"],
+        "Callback" => $values["callback"],
+        "Metadata" => [
```
This should be an empty array if Drupal doesn't need anything right now.

tamasd commented on modules/repo_sync/src/Form/RepoSyncDeleteForm.php:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  public function buildForm(array $form, FormStateInterface $form_state, $uuid = NULL) {
+    $form_state->set('uuid_tmp', $uuid);
```
This should be a `'#type' => 'value',` form field.

tamasd commented on modules/repo_sync/src/Service/Client.php:
```diff
@@ -189,11 +194,14 @@ protected function withCurl(string $method, string $url, array $headers, ?callab
-    $result = substr($rawresult, $header_size);
-    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-
+    if ($rawresult = curl_exec($ch)) {
```
This can be rewritten as:

    if (!($rawresult = curl_exec($ch))) {
      throw new Exception(...);
    }

I think it would also make sense to create an exception type for this, so it would be possible to store related metadata.

Commit:
Merge remote-tracking branch 'pronovix-devportal/8.x-2.x' into 8.x-2.x

Commit:
Additions and fixes

Added a view, and a callback controller; an update, and a delete form.
Fixed the changes requested to the pull request.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncViewController.php:
```diff
@@ -0,0 +1,56 @@
+      $this->messenger()->addError($e->getMessage());
+    }
+
+    $build = [
```
This should go inside the `try {}` block.

There should be some kind of an error page in the catch block. Maybe a standard one?

tamasd commented on modules/repo_sync/src/Controller/RepoSyncViewController.php:
```diff
@@ -0,0 +1,56 @@
+      }
+    }
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
I think this exception should be logged as well.

tamasd commented on modules/repo_sync/src/Form/RepoSyncCreateForm.php:
```diff
@@ -85,28 +79,34 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
-    catch (\Exception $e) {
-      $this->messenger->addError($e->getMessage());
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
This exception should be logged.

tamasd commented on modules/repo_sync/src/Form/RepoSyncDeleteForm.php:
```diff
@@ -31,24 +32,29 @@ public function getFormId() {
-    catch (\Exception $e) {
-      $this->messenger->addError($e->getMessage());
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
This exception should be logged.

tamasd commented on modules/repo_sync/src/Form/RepoSyncUpdateForm.php:
```diff
@@ -0,0 +1,132 @@
+      $result = json_decode(array_pop($result), TRUE);
+    }
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
This exception should be logged.

tamasd commented on modules/repo_sync/src/Form/RepoSyncUpdateForm.php:
```diff
@@ -0,0 +1,132 @@
+    ];
+    $form['result'] = [
+      '#type' => 'value',
+      '#value' => !empty($result) ? $result : NULL,
```
Why not `$result ?? NULL`?

tamasd commented on modules/repo_sync/src/Form/RepoSyncUpdateForm.php:
```diff
@@ -0,0 +1,132 @@
+      ]));
+    }
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
This exception should be logged.

tamasd commented on modules/repo_sync/src/Form/RepoSyncUpdateForm.php:
```diff
@@ -0,0 +1,132 @@
+
+    try {
+      $client("PUT", "/api/import/{$values["result"]["ID"]}", json_encode([
+        "ID" => $values["uuid"],
```
Instead of constructing the array here, I think it would be better to get the full `$form_state->getValue('result');`, and overwrite values in it. That way if an additional value is added, Drupal won't overwrite it.

tamasd commented on modules/repo_sync/src/Service/Client.php:
```diff
@@ -194,14 +197,14 @@ protected function withCurl(string $method, string $url, array $headers, ?callab
 
-    if ($rawresult = curl_exec($ch)) {
+    if (!($rawresult = curl_exec($ch))) {
+      throw new DevportalRepoSyncConnectionException('Curl handler error.');
```
This could be more verbose. Maybe use this: http://php.net/manual/en/function.curl-error.php

tamasd commented on modules/repo_sync/src/Controller/RepoSyncCollectionController.php:
```diff
@@ -36,36 +40,40 @@ public function content() {
-    catch (\Exception $e) {
-      $this->messenger->addError($e->getMessage());
+    catch (DevportalRepoSyncConnectionException $e) {
+      $this->messenger()->addError($e->getMessage());
```
This exception should be logged.

Commit:
Merge remote-tracking branch 'pronovix-devportal/8.x-2.x' into 8.x-2.x

Commit:
Replaced hook_install() with hook_modules_installed().

Config installer couldn't complete the installation with hook_install().
It works with hook_modules_installed().

Commit:
Fixed the changes requested to the pull request.

tamasd commented on modules/repo_sync/src/Controller/RepoSyncCallbackController.php:
```diff
@@ -0,0 +1,30 @@
+  public function callback(string $uuid, string $hash) {
+    return [
+      '#type' => 'markup',
+      '#markup' => $this->t("This is your UUID: $uuid, and this is your hash: $hash"),
```
This is a potential XSS here. Please use parameters for the `t()` function.

Commit:
Using parameters with the t() function.

Commit:
Create repo sync connector service

Create repo sync connector service. Change code so that it takes
advantage of the connector service.

Commit:
Merge remote-tracking branch 'pronovix-devportal/8.x-2.x' into 8.x-2.x

Commit:
Merge branch '8.x-2.x_repo-sync' into 8.x-2.x

Commit:
Removed duplicate code in devportal_api_referemce.module

tamasd commented on modules/repo_sync/src/Form/RepoSyncUpdateForm.php:
```diff
@@ -0,0 +1,124 @@
+   * @throws \Exception
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, $uuid = NULL) {
+    $result = $this->devportalRepoSyncConnection->getImport($uuid, TRUE);
```
Unnecessary second parameter.

Commit:
Remove unnecessary second commit.
